tests/Doc/QueryRangeExample.cs

```diff
@@ -0,0 +1,282 @@
+                    " comfortable ride for the bike path, the Summit gives a good value" +
+                    " for money.",
+                condition = "new",
+            },
+            new
+            {
+                model = "ThrillCycle",
+                brand = "BikeShind",
+                price = 815,
+                description = "An artsy,  retro-inspired bicycle that’s as functional as it is" +
+                    " pretty: The ThrillCycle steel frame offers a smooth ride. A" +
+                    " 9-speed drivetrain has enough gears for coasting in the city, but" +
+                    " we wouldn’t suggest taking it to the mountains. Fenders protect" +
+                    " you from mud, and a rear basket lets you transport groceries," +
+                    " flowers and books. The ThrillCycle comes with a limited lifetime" +
+                    " warranty, so this little guy will last you long past graduation.",
+                condition = "refurbished",
+            },
+        };
+
+        for (var i = 0; i < bicycles.Length; i++)
+        {
+            db.JSON().Set($"bicycle:{i}", "$", bicycles[i]);
+        }
+        // HIDE_END
+
+
+        // STEP_START range1
+        SearchResult res1 = db.FT().Search(
+            "idx:bicycle",
+            new Query("@price:[500 1000]")
+        );
+        Console.WriteLine(res1.TotalResults);   // >>> 3
+        // STEP_END
+
+        // Tests for 'range1' step.
+        // REMOVE_START
+        Assert.Equal(3, res1.TotalResults);
+        // REMOVE_END
+
+
+        // STEP_START range2
+        SearchResult res2 = db.FT().Search(
+            "idx:bicycle",
+            new Query().AddFilter(
+                new Query.NumericFilter("price", 500, 1000)
+            )
+        );
+        Console.WriteLine(res2.TotalResults);   // >>> 3
+        // STEP_END
+
+        // Tests for 'range2' step.
+        // REMOVE_START
+        Assert.Equal(3, res2.TotalResults);
+        // REMOVE_END
+
+
+        // STEP_START range3
+        SearchResult res3 = db.FT().Search(
+            "idx:bicycle",
+            new Query().AddFilter(
+                new Query.NumericFilter(
+                    "price",
+                    1000, true,
+                    double.PositiveInfinity, false
+                )
+            )
+        );
+        Console.WriteLine(res3.TotalResults);   // >>> 5
+        // STEP_END
+
+        // Tests for 'range3' step.
+        // REMOVE_START
+        Assert.Equal(5, res3.TotalResults);
+        // REMOVE_END
+
+
+        // STEP_START range4
+        SearchResult res4 = db.FT().Search(
+            "idx:bicycle",
+            new Query()
+                .AddFilter(
+                    new Query.NumericFilter(
+                        "price",
+                        double.NegativeInfinity, 2000
+                    )
+                )
+                .SetSortBy("price")
+                .Limit(0, 5)
+        );
```

Comment:
python example uses query string instead of numeric filter. 
it is better to cover more while keeping examples compatible across languages.
lets either pick numeric filter or query text and apply our choice to all (i did also notice that 'Redis CLI' example not uses filter as well).